torchtune/modules/attention.py

```diff
@@ -211,7 +211,7 @@ def forward(
         s_y = y.shape[1]
 
         if self.kv_cache and input_pos is None:
-            cache_size = self.kv_cache.size
+            cache_size = self.kv_cache.k_cache.dim(2)
```

Comment:
This is great! Much better than doing .item(). Can you keep self.size inside of the cache and do this change there? I think its a bit safer and cleaner.

FYI, a bit related to this PR: https://github.com/pytorch/torchtune/pull/1364

Reply:
Yes, good idea. I now just removed the item(), so size will be a tensor which can still be used in the same way as the scalar before.